src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js

```diff
@@ -852,6 +852,62 @@ ApiGlobe.prototype.removeEventListenerLayerChanged = function removeEventListene
     this.viewerDiv.removeEventListener('layerchanged:index', this.callbackLayerChanged, false);
 };
 
+/**
+ * Get the IPR (Intellectual Property Rights) of all layers in the scene.
+ * @constructor
+ * @return {array}  An array of IPR.
+ */
+
+ApiGlobe.prototype.getLayersIPR = function getLayersIPR() {
+    var IPR = [];
+    var colorLayers = this.scene.getMap().layersConfiguration.getColorLayers();
+    for (var i = 0; i < colorLayers.length; i++) {
+        IPR.push(colorLayers[i].options.ipr);
+    }
+    var elevationLayers = this.scene.getMap().layersConfiguration.getElevationLayers();
+    for (var j = 0; j < elevationLayers.length; j++) {
+        IPR.push(elevationLayers[j].options.ipr);
+    }
+    IPR = this.removeDuplicatesFromArray(IPR, 'name');
+    return IPR;
+};
+
+ApiGlobe.prototype.removeDuplicatesFromArray = function removeDuplicatesFromArray(originalArray, prop) {
+    var newArray = [];
+    var lookupObject = {};
+    for (var i in originalArray) {
+        lookupObject[originalArray[i][prop]] = originalArray[i];
+    }
+    for (i in lookupObject) {
+        newArray.push(lookupObject[i]);
+    }
+    return newArray;
+};
+
+/**
+ * Set the IPR for a layer.
+ * @constructor
+ * @param {param} Param - The id of the layer and the new IPR {id, ipr}.
+ */
+
+ApiGlobe.prototype.setLayerIPR = function setLayerIPR(param) {
```

Comment:
Note that I specified the attribut `param.ipr.name` mandatory in this method. If one choose to specify an IPR, this one must have one argument at least.